n_helpers.c

```diff
@@ -326,49 +210,143 @@ uint32_t NoteBinaryMaxDecodedLength(uint32_t bufferSize)
   @returns  The max required buffer size to hold the encoded data.
  */
 /**************************************************************************/
-uint32_t NoteBinaryMaxEncodedLength(uint32_t unencodedLength)
+uint32_t NoteBinaryCodecMaxEncodedLength(uint32_t unencodedLength)
 {
     return cobsEncodedMaxLength(unencodedLength);
 }
 
+//**************************************************************************/
+/*!
+  @brief  Get the length of the data stored on the Notecard. If there's no data
+          stored on the Notecard, then `*len` will return 0.
+
+  @param  len [out] The length of the decoded contents of the Notecard's binary
+          data store.
+
+  @returns An error string on error and NULL on success.
+ */
+/**************************************************************************/
+const char * NoteBinaryStoreDecodedLength(uint32_t *len)
+{
+    // Validate parameter(s)
+    if (!len) {
+        const char *err = ERRSTR("len cannot be NULL", c_bad);
+        NOTE_C_LOG_ERROR(err);
+        return err;
+    }
+
+    // Issue a "card.binary" request.
+    J *rsp = NoteRequestResponse(NoteNewRequest("card.binary"));
+    if (!rsp) {
+        const char *err = ERRSTR("unable to issue binary request", c_err);
+        NOTE_C_LOG_ERROR(err);
+        return err;
+    }
+
+    // Ensure the transaction doesn't return an error and confirm the binary
+    // feature is available.
+    if (NoteResponseError(rsp)) {
+        NOTE_C_LOG_ERROR(JGetString(rsp, "err"));
+        JDelete(rsp);
+        const char *err = ERRSTR("unexpected error received during handshake", c_err);
+        NOTE_C_LOG_ERROR(err);
+        return err;
+    }
+
+    // Examine "length" from the response to evaluate the length of the decoded
+    // data residing on the Notecard.
+    *len = JGetInt(rsp, "length");
+    JDelete(rsp);
+
+    return NULL;
+}
+
+//**************************************************************************/
+/*!
+  @brief  Get the required buffer length to receive the entire binary object
+          stored on the Notecard.
+
+  @param  len [out] The length required to hold the entire contents of the
+           Notecard's binary data store. If there's no data stored on the
```

Comment:
270-271 are indented by 1 additional space. This is really unacceptable, @zfields.

Reply:
:grimacing: